### PR TITLE
ABC-63: Enable release firmware image build

### PR DESCRIPTION
**Pull request by thomas-burley-bluefruit**
Main change here was switching from the build command:
`com.ti.ccstudio.apps.projectBuild`
to
`org.eclipse.cdt.managedbuilder.core.headlessbuild`

The new command more reliably builds the requested configuration. It also requires the project name to be specified, so an additional parameter for that has been added to the action.

### CODE REVIEW

File: .github/workflows/test_local_action.yml [hunk 1]
```diff
@@ -12,8 +12,30 @@ jobs:
...
+
+      - name: CCS Build action step - Debug
+        uses: ./
+        with:
+          project-path: 'test/working_ccs_project'
+          project-name: 'msp430_test_project'
+          build-configuration: Debug
+
+      - name: Upload release firmware artifact
+        uses: actions/upload-artifact@v4
```
Comment (grahame-student) [inline .github/workflows/test_local_action.yml:30]: prefer to use the full @x.y.z tag for actions as it makes workflows more deterministic

File: .github/workflows/test_local_action.yml [hunk 2]
```diff
@@ -12,8 +12,30 @@ jobs:
...
+        with:
+          name: 'Firmware artifact - Release'
+          path: test/working_ccs_project/Release/msp430_test_project.txt
+          if-no-files-found: error
+
+      - name: Upload debug firmware artifact
+        uses: actions/upload-artifact@v4
```
Comment (grahame-student) [inline .github/workflows/test_local_action.yml:37]: prefer to use the full @x.y.z tag for actions as it makes workflows more deterministic

File: .github/workflows/test_local_action.yml [hunk 3]
```diff
@@ -12,8 +12,30 @@ jobs:
...
+      - name: CCS Build action step - Debug
+        uses: ./
+        with:
+          project-path: 'test/working_ccs_project'
+          project-name: 'msp430_test_project'
+          build-configuration: Debug
+
+      - name: Upload release firmware artifact
+        uses: actions/upload-artifact@v4
+        with:
```
Comment (grahame-student) [inline .github/workflows/test_local_action.yml:31]: You can maximise the compression level for the ti-txt files
https://github.com/actions/upload-artifact?tab=readme-ov-file#altering-compressions-level-speed-v-size

File: .github/workflows/test_local_action.yml [hunk 4]
```diff
@@ -12,8 +12,30 @@ jobs:
...
+          name: 'Firmware artifact - Release'
+          path: test/working_ccs_project/Release/msp430_test_project.txt
+          if-no-files-found: error
+
+      - name: Upload debug firmware artifact
+        uses: actions/upload-artifact@v4
+        with:
```
Comment (grahame-student) [inline .github/workflows/test_local_action.yml:38]: You can maximise the compression level for the ti-txt files
https://github.com/actions/upload-artifact?tab=readme-ov-file#altering-compressions-level-speed-v-size

File: .github/workflows/test_released_action.yml [hunk 5]
```diff
@@ -19,7 +19,30 @@ jobs:
...
+
+      - name: CCS Build action step - Debug
+        uses: apollo-fire/ccs-build@v0.0.6
+        with:
+          project-path: 'test/working_ccs_project'
+          project-name: 'msp430_test_project'
+          build-configuration: Debug
+
+      - name: Upload release firmware artifact
+        uses: actions/upload-artifact@v4
```
Comment (grahame-student) [inline .github/workflows/test_released_action.yml:37]: prefer to use the full @x.y.z tag for actions as it makes workflows more deterministic

File: .github/workflows/test_released_action.yml [hunk 6]
```diff
@@ -19,7 +19,30 @@ jobs:
...
+        with:
+          name: 'Firmware artifact - Release'
+          path: test/working_ccs_project/Release/msp430_test_project.txt
+          if-no-files-found: error
+
+      - name: Upload debug firmware artifact
+        uses: actions/upload-artifact@v4
```
Comment (grahame-student) [inline .github/workflows/test_released_action.yml:44]: prefer to use the full @x.y.z tag for actions as it makes workflows more deterministic

File: .github/workflows/test_released_action.yml [hunk 7]
```diff
@@ -19,7 +19,30 @@ jobs:
...
+      - name: CCS Build action step - Debug
+        uses: apollo-fire/ccs-build@v0.0.6
+        with:
+          project-path: 'test/working_ccs_project'
+          project-name: 'msp430_test_project'
+          build-configuration: Debug
+
+      - name: Upload release firmware artifact
+        uses: actions/upload-artifact@v4
+        with:
```
Comment (grahame-student) [inline .github/workflows/test_released_action.yml:38]: You can maximise the compression level for the ti-txt files
https://github.com/actions/upload-artifact?tab=readme-ov-file#altering-compressions-level-speed-v-size

File: .github/workflows/test_released_action.yml [hunk 8]
```diff
@@ -19,7 +19,30 @@ jobs:
...
+          name: 'Firmware artifact - Release'
+          path: test/working_ccs_project/Release/msp430_test_project.txt
+          if-no-files-found: error
+
+      - name: Upload debug firmware artifact
+        uses: actions/upload-artifact@v4
+        with:
```
Comment (grahame-student) [inline .github/workflows/test_released_action.yml:45]: You can maximise the compression level for the ti-txt files
https://github.com/actions/upload-artifact?tab=readme-ov-file#altering-compressions-level-speed-v-size

File: README.md
Comment (grahame-student) [inline README.md:1]: bump the version numbers to v0.0.6 too

File: action.yml
Comment (grahame-student) [inline action.yml:1]: the action.yml files in the following locations also need updating:
* https://github.com/apollo-fire/ccs-build/tree/abc-63-enable-release-firmware-build/compiler/18.1.3
* https://github.com/apollo-fire/ccs-build/tree/abc-63-enable-release-firmware-build/compiler/21.6.0

**Review by grahame-student — CHANGES_REQUESTED** [review]
Mostly there, only a couple of small tweaks required

File: .github/workflows/test_local_action.yml [hunk 9]
```diff
@@ -27,15 +27,17 @@ jobs:
           build-configuration: Debug
 
       - name: Upload release firmware artifact
-        uses: actions/upload-artifact@v4
+        uses: actions/upload-artifact@v4.3.0
         with:
           name: 'Firmware artifact - Release'
           path: test/working_ccs_project/Release/msp430_test_project.txt
           if-no-files-found: error
+          compression-level: 6
```
Comment (grahame-student) [inline .github/workflows/test_local_action.yml:35]: max compression is 9

File: .github/workflows/test_local_action.yml [hunk 10]
```diff
@@ -12,8 +12,32 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4.1.1
 
-      - name: CCS Build action step
+      - name: CCS Build action step  - Release
```
Comment (grahame-student) [inline .github/workflows/test_local_action.yml:15]:
```suggestion
      - name: CCS Build action step - Release
```

**Review by grahame-student — APPROVED** [review]
LGTM

**Comment by grahame-student** [comment]
Just had a thought, will this cause workflows transitioning from the previous version to stop working?

**Comment by thomas-burley-bluefruit** [comment]
> @thomas-burley-bluefruit Just had a thought, will this cause workflows transitioning from the previous version to stop working?
> 
> If so then semver (https://semver.org/) tells us we should bump the major version number.

Yes it will because there is a new mandatory input on the action. So it should be 1.0.0 now if I'm reading that page correctly?

**Review by ThatTallGuy14 — APPROVED** [review]
LGTM

**Review by grahame-student — APPROVED** [review]
LGTM

**Pull request merged**